Sources/Core/GTMSessionFetcherLogging.m

```diff
@@ -75,141 +77,165 @@ @implementation GTMSessionFetcher (GTMSessionFetcherLogging)
 static NSString *gLoggingProcessName = nil;
 
 + (void)setLoggingDirectory:(NSString *)path {
-  gLoggingDirectoryPath = [path copy];
+  @synchronized([GTMSessionFetcher class]) {
+    gLoggingDirectoryPath = [path copy];
+  }
 }
 
 + (NSString *)loggingDirectory {
-  if (!gLoggingDirectoryPath) {
-    NSArray *paths = nil;
+  @synchronized([GTMSessionFetcher class]) {
+    if (!gLoggingDirectoryPath) {
+      NSArray *paths = nil;
 #if TARGET_IPHONE_SIMULATOR
-    // default to a directory called GTMHTTPDebugLogs into a sandbox-safe
-    // directory that a developer can find easily, the application home
-    paths = @[ NSHomeDirectory() ];
+      // default to a directory called GTMHTTPDebugLogs into a sandbox-safe
+      // directory that a developer can find easily, the application home
+      paths = @[ NSHomeDirectory() ];
 #elif TARGET_OS_IPHONE
-    // Neither ~/Desktop nor ~/Home is writable on an actual iOS, watchOS, or tvOS device.
-    // Put it in ~/Documents.
-    paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+      // Neither ~/Desktop nor ~/Home is writable on an actual iOS, watchOS, or tvOS device.
+      // Put it in ~/Documents.
+      paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
 #else
-    // default to a directory called GTMHTTPDebugLogs in the desktop folder
-    paths = NSSearchPathForDirectoriesInDomains(NSDesktopDirectory, NSUserDomainMask, YES);
+      // default to a directory called GTMHTTPDebugLogs in the desktop folder
+      paths = NSSearchPathForDirectoriesInDomains(NSDesktopDirectory, NSUserDomainMask, YES);
 #endif
 
-    NSString *desktopPath = paths.firstObject;
-    if (desktopPath) {
-      NSString *const kGTMLogFolderName = @"GTMHTTPDebugLogs";
-      NSString *logsFolderPath = [desktopPath stringByAppendingPathComponent:kGTMLogFolderName];
+      NSString *desktopPath = paths.firstObject;
+      if (desktopPath) {
+        NSString *const kGTMLogFolderName = @"GTMHTTPDebugLogs";
+        NSString *logsFolderPath = [desktopPath stringByAppendingPathComponent:kGTMLogFolderName];
+
+        NSFileManager *fileMgr = [NSFileManager defaultManager];
+        BOOL isDir;
+        BOOL doesFolderExist = [fileMgr fileExistsAtPath:logsFolderPath isDirectory:&isDir];
+        if (!doesFolderExist) {
+          // make the directory
+          doesFolderExist = [fileMgr createDirectoryAtPath:logsFolderPath
+                               withIntermediateDirectories:YES
+                                                attributes:nil
+                                                     error:NULL];
+          if (doesFolderExist) {
```

Comment:
What are we doing in the unlikely case that this fails?

Reply:
Same.